src/lib/interactionManager/types.ts

```diff
@@ -0,0 +1,21 @@
+import { Interaction } from './interaction'
+import { CredentialOfferFlow } from './credentialOfferFlow'
+import { CredentialOffer } from 'jolocom-lib/js/interactionTokens/interactionTokens.types'
+import { SignedCredential } from 'jolocom-lib/js/credentials/signedCredential/signedCredential'
+import { CredentialRequestFlow } from './credentialRequestFlow'
+
+export interface InteractionState {
+  [nonce: string]: Interaction
+}
+
+export type InteractionFlows = CredentialOfferFlow | CredentialRequestFlow
```

Comment:
hey looky here 👀 the whole interaction tokens (/JWTs) typing dilemma all over again. #fullcircle 
It's the first task I ever did at jolocom, try to rewrite this bit: https://github.com/jolocom/jolocom-lib/blob/develop/ts/interactionTokens/JSONWebToken.ts#L29

Reply:
I think a better approach will be to create a possibly empty base class (or interface (or both)) `InteractionFlow`, kinda like an anchor for this type. Just inherit from that and use it for your typing requirements.